src/toxcore/dht/dht_node.rs

```diff
@@ -18,35 +18,71 @@
     along with Tox.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-
-/*!
-Manage ping_id.
-Generate ping_id on request packet, check ping_id on response packet.
+/*! Data structure used by Bucket.
+PackedNode type contains PK and SocketAddress.
+PackedNode does not contain status of Node, this struct contains status of node.
+Bucket needs status of node, because BAD status node should be replaced with higher proirity than GOOD node.
+Even GOOD node is farther than BAD node, BAD node should be replaced.
+Here, GOOD node is the node responded within 162 seconds, BAD node is the node not responded over 162 seconds.
 */
 
+use std::net::SocketAddr;
 use std::collections::HashMap;
 use std::time::{Duration, Instant};
-
 use toxcore::crypto_core::*;
+use toxcore::dht::packed_node::*;
+
+/** Status of node in bucket.
+Good means it is online and responded within 162 seconds
+Bad means it is probably offline and did not responded for over 162 seconds
+When new peer is added to bucket, Bad status node should be replace.
+If there are no Bad nodes in bucket, node which is farther than peer is replaced.
+
+Manage ping_id.
+Generate ping_id on request packet, check ping_id on response packet.
+*/
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum NodeStatus {
+    /// online
+    Good,
+    /// maybe offline
+    Bad,
+}
 
-/// peer info.
-#[derive(Clone, Debug)]
-pub struct ClientData {
+/** Struct used by Bucket, DHT maintains close node list, when we got new node,
+we should make decision to add new node to close node list, or not.
+the PK's distance and status of node help making decision.
+Bad node have higher priority than Good node.
+If both node is Good node, then we compare PK's distance.
+
+Generate ping_id on request packet, check ping_id on response packet.
+*/
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct DhtNode {
+    /// Socket addr of node.
+    pub saddr: SocketAddr,
+    /// Public Key of the node.
+    pub pk: PublicKey,
+    /// Status of node
+    pub status: NodeStatus,
```

Comment:
It seems we can have method that returns status instead of storing it as a field because this status uniquely defined by `last_resp_time`.